backend/internal/application/service/applicationservice.go

```diff
@@ -129,11 +129,12 @@ func (as *ApplicationService) CreateApplication(app *model.ApplicationDTO) (*mod
 		Token:                     rootToken,
 	}
 	if inboundAuthConfig != nil {
-		// Wrap the finalOAuthToken in OAuthTokenConfig structure
+		// Wrap the finalOAuthAccessToken and finalOAuthIDToken in OAuthTokenConfig structure
 		var oAuthTokenConfig *model.OAuthTokenConfig
-		if finalOAuthToken != nil {
+		if finalOAuthAccessToken != nil || finalOAuthIDToken != nil {
```

Comment:
We should have separate if conditions for this right?
Let's check whether we actually need this if condition(s)